### ISSUE TITLE
Configuring google-oauth-plugin credentials

### ISSUE THREAD

**Issue by jpigree** [jenkinsci/configuration-as-code-plugin#785]
Informations:
- [ Jenkins ver. 2.164 ] Jenkins version
- [ 1.8 ] Plugin version
- [ Official Jenkins Docker image ] OS

Hi, I recently used JCasC plugin successfully to replace hundreds of groovy initialization lines and I am pretty happy with it. :)

However, I have trouble translating some specific credential from the google-oauth-plugin(https://github.com/jenkinsci/google-oauth-plugin) into JCasC definition.

So far, I got to this point:
```yaml
credentials:
  system:
    domainCredentials:
      - credentials:
        - googleRobotPrivateKey:
          projectId: "<PROJECT ID>"
          serviceAccountConfig:
            json:
              prevJsonKeyFile: '<THE KEY AS A STRING>'
```

But it just takes the string as the file name (prevJsonKeyFile).

I found an interesting post from a guy which does this in groovy init script:
=> https://github.com/jenkinsci/google-oauth-plugin/issues/5#issuecomment-393128560

From what I understand, the plugin needs a FileStream to actually process my credential even if, in the JCasC documentation generated on my jenkins instance, it requires a string.

![JCasC_API_doc](https://user-images.githubusercontent.com/3977280/54860070-e5867b00-4cb8-11e9-814b-079da53e55c5.png)

I didn't found any examples on how to pass on a credential as a file stream with JCasC. 

Can someone help me on how to do it? I will be glad to contribute back an example for this use-case when it will work.

Thanks!!!

**Comment by timja** [comment]
Not sure if its possible currently, I would create an issue on their side about using standard databinding techniques

**Comment by jpigree** [comment]
Did you already create an issue on their side? I can do it if you don't have the time.
But perhaps, it won't have the same impact if it is a maintainer of the JCasC plugin doing it.

Thank you for your help. :)

**Comment by timja** [comment]
No I haven't, it's better if a user of the plugin does it, I'll subscribe to it though

**Comment by timja** [comment]
No I haven't, it's better if a user of the plugin does it, I'll subscribe to it though

**Comment by casz** [comment]
I am just linking. 😁
https://github.com/jenkinsci/google-oauth-plugin/pull/19 should fix your issue

**Comment by stephenashank** [comment]
@casz @jpigree I am happy to report that we have just released [google-oauth-plugin-0.8]( https://github.com/jenkinsci/google-oauth-plugin/releases/tag/google-oauth-plugin-0.8) with support for CasC.

**Comment by jpigree** [comment]
Awesome. Thanks for the release and the information @stephenashank. Updating right away. :)
